src/Crowdsale/DynamicPriceSale.sol

```diff
@@ -136,16 +139,29 @@ contract DynamicSale is ContextUpgradeable, ReentrancyGuardUpgradeable, Ownable2
         uint256 currentSupply = token.totalSupply();
         require(currentSupply >= priceCurveMinValue, "DynamicSale: current totalSupply too low");
         require(currentSupply + amount <= priceCurveMaxValue, "DynamicSale: totalSupply limit reached");
-        /// @dev sale-function coefficients
-        uint256 c = 420;
-        uint256 b = 32000461777723 * (10**54);
-        uint256 a = 11680057722 * (10**36);
-
-        uint256 start = currentSupply;
-        uint256 end = start + amount;
-        newPrice = c - a / end**2 + b / end**3;
-        totalCost = c * (end - start) + a * (1 / end - 1 / start) - (b / 2) * (1 / end**2 - 1 / start**2);
+        // /// @dev sale-function coefficients
+
+        int256 c = 420;
+        int256 b = 32000461777723 * (10**54);
+        int256 a = 11680057722 * (10**36);
+
+        // Get current supply
+        int256 supplyBeforeBuy = int256(currentSupply);
+        // Calculate supply after buying
+        int256 supplyAfterBuy = supplyBeforeBuy + int256(amount);
+        // Get unit price after amount has been bought acoording to formula: c + a/S^2 + b/S^3
+        int256 _newPrice = c - (a / supplyAfterBuy**2) + (b / supplyAfterBuy**3);
```

Comment:
I think it makes sense to implement a view function to get the current price. That is a good suggestion and saves on gas. One thing though, the contract has a function called `setNewPrice()`, which writes to the state variable. This seems however only be able to be useful before the actual sale happens, because when an actual buy happens, it always calculates the cost based on the formula, and not on the state variable.

I think it would make sense to also to rename the state variable and function to something like `current...` -> `initial...`. Wdyt? @fabianschu